cassandra/src/main/java/org/hibernate/ogm/datastore/cassandra/impl/CassandraDatastoreProvider.java

```diff
@@ -0,0 +1,266 @@
+/*
+ * Hibernate OGM, Domain model persistence for NoSQL datastores
+ *
+ * License: GNU Lesser General Public License (LGPL), version 2.1 or later
+ * See the lgpl.txt file in the root directory or <http://www.gnu.org/licenses/lgpl-2.1.html>.
+ */
+package org.hibernate.ogm.datastore.cassandra.impl;
+
+import java.util.Collection;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+
+import com.datastax.driver.core.Cluster;
+import com.datastax.driver.core.ResultSet;
+import com.datastax.driver.core.Row;
+import com.datastax.driver.core.Session;
+
+import com.datastax.driver.core.exceptions.DriverException;
+
+import org.hibernate.boot.registry.classloading.spi.ClassLoaderService;
+
+import org.hibernate.mapping.Table;
+
+import org.hibernate.ogm.datastore.cassandra.CassandraDialect;
+import org.hibernate.ogm.datastore.cassandra.impl.configuration.CassandraConfiguration;
+import org.hibernate.ogm.datastore.spi.BaseDatastoreProvider;
+import org.hibernate.ogm.datastore.spi.SchemaDefiner;
+import org.hibernate.ogm.dialect.spi.GridDialect;
+import org.hibernate.ogm.model.key.spi.AssociationKeyMetadata;
+import org.hibernate.ogm.options.spi.OptionsService;
+import org.hibernate.ogm.persister.impl.OgmCollectionPersister;
+import org.hibernate.ogm.util.configurationreader.spi.ConfigurationPropertyReader;
+
+import org.hibernate.persister.collection.CollectionPersister;
+
+import org.hibernate.ogm.datastore.cassandra.logging.impl.Log;
+import org.hibernate.ogm.datastore.cassandra.logging.impl.LoggerFactory;
+
+import org.hibernate.service.spi.Configurable;
+import org.hibernate.service.spi.ServiceRegistryAwareService;
+import org.hibernate.service.spi.ServiceRegistryImplementor;
+import org.hibernate.service.spi.Startable;
+import org.hibernate.service.spi.Stoppable;
+
+/**
+ * Datastore service layered on Cassandra's java-driver i.e. CQL3 native transport.
+ *
+ * @author Jonathan Halliday
+ */
+public class CassandraDatastoreProvider extends BaseDatastoreProvider
+		implements Startable, Stoppable, ServiceRegistryAwareService, Configurable {
+
+	private static final Log log = LoggerFactory.getLogger();
+
+	private ServiceRegistryImplementor serviceRegistry;
+
+	private CassandraConfiguration config;
+
+	private Cluster cluster;
+	private Session session;
```

Comment:
hmm, weird, the driver should deal with that since it is designed for talking to a cluster in which nodes come and go routinely. Maybe some edge case in which it's assuming the cluster as a whole is robust enough that at least one node is reachable.